tests/acceptance/stepDefinitions/filesContext.js

```diff
@@ -229,9 +244,109 @@ Then(/there should be (\d+) files\/folders listed on the webUI/, async function
   return client.assert.equal(allFileRows.value.length, noOfItems)
 })
 
-const assertDeletedElementsAreNotListed = function () {
-  for (const element of deletedElementsTable) {
+const assertElementsAreListed = function (elements) {
+  for (const element of elements) {
+    client.page.FilesPageElement.filesList().assertElementListed(element)
+  }
+  return client
+}
+
+const assertElementsAreNotListed = function (elements) {
+  for (const element of elements) {
     client.page.FilesPageElement.filesList().assertElementNotListed(element)
   }
   return client
 }
+
+const assertDeletedElementsAreNotListed = function () {
+  return assertElementsAreNotListed(deletedElementsTable)
+}
+
+const assertDeletedElementsAreListed = function () {
+  return assertElementsAreListed(deletedElementsTable)
+}
+
+/**
+ * Get the list of trashbin items for a user
+ * in following format
+ * [{
+ *  "href":
+ *  "originalFilename":
+ *  "originalLocation":
+ *  "deleteTimestamp":
+ *  "lastModified":
+ * },...]
+ *
+ * @param {string} user
+ */
+const getTrashBinElements = function (user) {
+  return new Promise((resolve) => {
+    webdav.propfind(`/trash-bin/${user}`, user, userSettings.getPasswordForUser(user),
+      [
+        'oc:trashbin-original-filename',
+        'oc:trashbin-original-location',
+        'oc:trashbin-delete-timestamp',
+        'd:getlastmodified'
+      ])
+      .then(str => {
+        let trashData = JSON.parse(convert.xml2json(str, { compact: true }))['d:multistatus']['d:response']
+        let trashItems = []
+        trashData.map(trash => {
+          trashItems.push({
+            href: trash['d:href']._text,
+            originalFilename: trash['d:propstat']['d:prop']['oc:trashbin-original-filename']._text,
+            originalLocation: trash['d:propstat']['d:prop']['oc:trashbin-original-location']._text,
+            deleteTimestamp: trash['d:propstat']['d:prop']['oc:trashbin-delete-timestamp']._text,
+            lastModified: trash['d:propstat']['d:prop']['d:getlastmodified']._text
+          })
+        })
+        resolve(trashItems)
+      })
+  })
+}
+
+Then('as {string} file/folder {string} should exist in trash', function (user, element) {
```

Comment:
these steps do only WEBDAV requests and check them, so lets keep them in WebDav context or even a separate context, to keep every file as short as possible